src/pages/endorsement-questionnaire.js

```diff
@@ -93,20 +92,17 @@ const Endorsements = () => {
 
   const handleSubmit = (e) => {
     e.preventDefault();
+    e.persist();
 
+    setSubmitted({ status: 'pending' });
     client.getSpace(data.site.siteMetadata.tokens.spaceId)
       .then((space) => space.getEnvironment('master'))
-      .then((environment) => {
-        setSubmitted({ status: 'pending' });
-        return environment.createEntry('candidateQuestionnaires', contentfulize(questionnaire));
-      })
-      .then((res) => {
-        if (res.fields) {
-          setSubmitted({ status: 'submitted' });
-        } else {
-          setSubmitted({ status: 'trouble' });
-        }
-      }, (err) => { discard(err); setSubmitted({ status: 'trouble' }); });
+      .then((environment) => environment.createEntry('candidateQuestionnaires', contentfulize(questionnaire)))
```

Comment:
We should still set `{status: 'pending' }`, shouldn't we? otherwise this looks great

Reply:
Actually "Pending" gets set as soon as we start the form submission, in line 97. Then gets overwritten with either a success or a failure status!